### PR TITLE
Refactor on-core unit-tests

### PR DESCRIPTION
**Pull request by jlongever**
- Update scheduler-spec, events-spec, task-graph-runner-spec
- Remove rabbitmq-server dependency.

@RackHD/corecommitters @keedya @tannoa2 @uppalk1 @zyoung51

### CODE REVIEW

File: spec/lib/protocol/events-spec.js [hunk 1]
```diff
@@ -4,286 +4,231 @@
...
-                } catch (err) {
-                    done(err);
-                }
+            messenger.subscribe = sinon.spy(function(a,b,callback) {
+                callback(data,testMessage);
+                return Promise.resolve(testSubscription);
+            });
+            messenger.request.resolves(data);
+            
+            events.subscribeTftpSuccess(nodeId, function (_data) {
```
Comment (zyoung51) [inline spec/lib/protocol/events-spec.js]: The promise should be returned or done () should be called.

File: spec/lib/protocol/events-spec.js [hunk 2]
```diff
@@ -4,286 +4,231 @@
...
-                    done();
-                } catch (err) {
-                    done(err);
-                }
+            messenger.subscribe = sinon.spy(function(a,b,callback) {
+                callback(data,testMessage);
+                return Promise.resolve(testSubscription);
+            });
+            messenger.request.resolves(data);
+            events.subscribeTftpFailure(nodeId, function (_data) {
```
Comment (zyoung51) [inline spec/lib/protocol/events-spec.js]: Return the promise

**Comment by jlongever** [comment]
Thank you sir, will fix it.

**Comment by yyscamper** [comment]
:+1: 
@jlongever : thanks for this fix. Whether it is means we don't need to close all background (on-http/on-taskgraph)  services before running the on-core unittest

**Comment by keedya** [comment]
:+1

**Comment by jlongever** [comment]
test this please

**Comment by jlongever** [comment]
Closing this PR,  collapsed changes into PR #74

**Pull request closed without merging**